### ISSUE TITLE
showWindows expects NSURL incorrectly

### ISSUE THREAD

**Issue by Timmmm** [gitx/gitx#152]
If you drag/drop a directory onto GitX that is already open it will get to `showWindows()`:

```
- (void)showWindows
{
	NSScriptCommand *command = [NSScriptCommand currentCommand];

	if (command) {
		NSURL *repoURL = [command directParameter];

		// on app launch there may be many repositories opening, so double check that this is the right repo
		if (repoURL) {
			NSLog(@"RepoURL is not null");
			repoURL = [PBRepositoryFinder gitDirForURL:repoURL];
			if ([repoURL isEqual:self.fileURL]) {
				NSArray *arguments = command.arguments[@"openOptions"];
				[self handleGitXScriptingArguments:arguments];
			}
		}
	}

	[[[self windowController] window] setTitle:[self displayName]];

	[super showWindows];
}
```

It expects `repoURL` to be an `NSURL*`, but actually it is an `NSMutableArray`. This means that later stuff fails and it never gets to `[super showWindows]` and shows that window that is already open.

I couldn't actually find any instance of `directParameter` returning an `NSURL*`. Are you sure this code was ever correct?

**Comment by tiennou** [comment]
This part of the code "handles" arguments from `gitx`, so if you `gitx --local` in an opened repository, you'll get the "correct" path  (which TBH was fixed by 00b90decb577c3b55a57235a701721c79c56c4d7). The check is there in case you're causing GitX to launch the first time ("many repositories opening").

~I'm confused as to how you ended up with a mutable array. I've tested one d&d of the currently opened repo, and got a nil `command`.~ Multiple repoes was the correct answer 😉.
